### PR TITLE
Assignment1

### PR DESCRIPTION
**Pull request by megha2627**
Solution for  problems  in C

### CODE REVIEW

File: asign1.c [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+#include<stdio.h>
+#include<ctype.h>
+#include <string.h>
+int main(){
+    char str[1000];
+    printf("enter the string of expression using only +,-,*,/ operators:\n");
+    fgets(str, sizeof(str), stdin);
+    str[strcspn(str, "\n")] = 0; 
+
+    int p=strlen(str);
```
Comment (intimetec-shiv-maheshwari) [inline asign1.c:52]: Do not use predefined functions, Instead of this, create your own function to achieve the similar functionality

File: asign1.c [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    int curr=0;
```
Comment (intimetec-shiv-maheshwari) [inline asign1.c:53]: Please use meaningful names when you declare a variable. Keep this comment in consideration wherever you have done the variable declaration.

File: asign1.c [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    int last = 0;
+    
+    char sign = '+';
+    int result = 0;
+    for (int i = 0; i < p;i++){ 
+        char chr = str[i];
+        if(isdigit(chr)){
```
Comment (intimetec-shiv-maheshwari) [inline asign1.c:60]: Use unsigned char when passing to isspace/isdigit to avoid UB.

File: asign1.c [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+                
+            
+        }
+       
+       
+    }
+    result=result+last;
+    printf("%d\n", result);
+    return 0;
+}
```
Comment (intimetec-shiv-maheshwari) [inline asign1.c:107]: You can segregate the logic into a function and then use that function for better readability 

Additionally, when given this input - 3 ++ 2 
Code fails silently with no error message

File: asign2.c [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+
+void addUser() {
+    FILE *fp = fopen("users.txt", "a");
+    if (fp == NULL) {
+        printf("Could not open file.\n");
+        return;
+    }
+    struct User u;
+    printf("Enter ID: ");
+    scanf("%d", &u.id);
```
Comment (intimetec-shiv-maheshwari) [inline asign2.c:53]: Test your code with input - 'i' as user id, the code fails 
Handle these type of input corner cases as well

File: asign1.c (hunk 2 above)
Comment (megha2627) [reply asign1.c]: I named all the variables in camelCase  now

File: asign1.c (hunk 3 above)
Comment (megha2627) [reply asign1.c]: I use unsigned char now

File: asign1.c (hunk 4 above)
Comment (megha2627) [reply asign1.c]: segregate the logic into a function in that part i am currently working

File: asign2.c [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+#include <string.h>
+
+struct User {
+    int id;
+    char name[50];
+    int age;
+};
+
+void addUser() {
+    FILE *fp = fopen("users.txt", "a");
```
[the remark this answers is not in the document]
Comment (megha2627) [reply asign2.c]: i check firstly if the id exist or not if it exist ask user for another id in the updated code

File: asign2.c [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+    while (1) {
+        printf("\nMenu\n");
+        printf("1. Add User\n");
+        printf("2. Show Users\n");
+        printf("3. Update User\n");
+        printf("4. Delete User\n");
+        printf("5. Exit\n");
+        printf("Choice: ");
+        scanf("%d", &ch);
+
```
[the remark this answers is not in the document]
Comment (megha2627) [reply asign2.c:226]: i handle this type of corner case by create function getInput in updated code
